pilot/pkg/model/push_context.go

```diff
@@ -310,12 +311,29 @@ func (ps *PushContext) UpdateMetrics() {
 }
 
 // Services returns the list of services that are visible to a Proxy in a given config namespace
-func (ps *PushContext) Services(proxy *Proxy) []*Service {
+func (ps *PushContext) Services(p *Proxy) []*Service {
 	// TODO: use network scopes here, and return services in the proxy namespace if
 	// configured to use current_namespace as the scope and proxy is not nil
-	//if proxy == nil {
-	//	return ps.allServices
-	//}
+	if p == nil {
+		return ps.allServices
+	}
+	p.mutex.RLock()
+	defer p.mutex.RUnlock()
+	if p.serviceDependencies != nil {
+		return p.serviceDependencies
+	}
```

Comment:
this is not used and its unclear how this will be populated (not to mention the memory usage to store array of pointers per proxy, to same set of objects). Can you remove this in this iteration please? If we cant get the namespace maps working, we can add this code back, but I suspect we will then have a plethora of objects like p.virtualServices, p.destinationRules, and so on..

Reply:
Sure, I can remove this, agree 1.1 could just do namespace deps.

But AFAIK the API have service names as well.